### ISSUE TITLE
Cordova - Multiple dex files define 

### ISSUE THREAD

**Issue by Divyaadz** [arnesson/cordova-plugin-firebase#32]
I like to use Fibebase Analytic in my cordova project, while trying to build android below error is thrown. Im using jdk 1.8

Please let em know what is the issue and how to resolve it. Thanks

Error: cmd: Command failed with exit code 1 Error output:
Dex: Error converting bytecode to dex:
Cause: com.android.dex.DexException: Multiple dex files define Lcom/google/firebase/FirebaseException;
    UNEXPECTED TOP-LEVEL EXCEPTION:
com.android.dex.DexException: Multiple dex files define Lcom/google/firebase/FirebaseException;
        at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:591)
        at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:546)
        at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:528)
        at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:164)
        at com.android.dx.merge.DexMerger.merge(DexMerger.java:188)
        at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:504)
        at com.android.dx.command.dexer.Main.runMonoDex(Main.java:334)
        at com.android.dx.command.dexer.Main.run(Main.java:277)
        at com.android.dx.command.dexer.Main.main(Main.java:245)
        at com.android.dx.command.Main.main(Main.java:106)

**Comment by devthree** [comment]
I got this error too before. From what I learn, I got this error because I use cordova plug-in that use different android support and play services version. I solve this error by open build.gradle in /platforms/android, and add this code after buildscript : 

configurations {
all*.exclude group: 'com.android.support', module: 'support-v4'
all*.exclude group: 'com.google.android.gms', module: 'play-services'
}

I don't know if this good solution since build.gradle is auto generated by cordova, but by adding that code cordova forced to use android support and play services that I set in configurations. Hope it help you.

**Comment by Chuckv01** [comment]
I was running into this too. This should be fixed in the latest commit that pulled in the fix.

**Comment by Divyaadz** [comment]
I also face same error.

**Comment by robertarnesson** [comment]
are you still getting this with cordova-plugin-firebase 0.1.10 ?

**Comment by srkgupta** [comment]
I still face the same error with cordova-plugin-firebase 0.1.18? Any ideas, how do we fix this? Tried a workaround suggested by @devthree, but didn't work. I still get the exact same exception as mentioned in this issue.

**Comment by deldreth** [comment]
I encountered the same issue after integrating https://github.com/mauron85/react-native-background-geolocation into my project. If you're using gradle you can check the dependency tree with something like `gradle -q dependencies app:dependencies`. I noticed that where firebase was included it being set to version `9.8.0`.

My solution was to modify my app's dependency compile of `com.google.android.gms:play-services-gcm` from version `9.0.0` to `9.8.0`.

So in my app/build.gradle dependecies...
```
dependencies {
...
compile ('com.google.android.gms:play-services-gcm:9.8.0')
}
```

Hopefully this information will assist.
